its/src/test/java/com/sonar/maven/it/suite/JavaTest.java

```diff
@@ -120,6 +120,24 @@ void setJavaVersionProperties() throws IOException {
       entry("sonar.java.target", "1.8"));
   }
 
+  @Test
+  void extractJdkHomeFromJre() throws IOException {
+    File outputProps = temp.resolve("out.properties").toFile();
+    outputProps.createNewFile();
+
+    File pom = ItUtils.locateProjectPom("jdkHome/extractFromJre");
+    MavenBuild build = MavenBuild.create(pom)
+      .setGoals(sonarGoal())
+      .setProperty("sonar.scanner.internal.dumpToFile", outputProps.getAbsolutePath());
+    executeBuildAndValidateWithoutCE(build);
+
+    Properties props = getProps(outputProps);
+    Object value = props.getOrDefault("sonar.java.jdkHome", "");
+    if (value instanceof String) {
+      assertThat((String) value).isNotEmpty();
+    }
```

Comment:
We can probably "force-cast" the value behind `sonar.java.jdkHome` as a string. If we encounter any other type, we probably want to find out at testing time rather than later. And if there is no value, then we should find out as well.
```suggestion
    String value = props.getProperty("sonar.java.jdkHome");
    assertThat(value).isNotEmpty();
```

Reply:
Test against JVM running JUnit test

Reply:
:+1: